### ISSUE TITLE
`scrollable-region-focusable` conflicts with `jsx-a11y/no-noninteractive-tabindex` eslint rule

### ISSUE THREAD

**Issue by vincerubinetti** [dequelabs/axe-core#4566]
### Product

axe-core

### Question

I've encountered an Axe error on a block of code in a wide `<pre>` element with `overflow-x: auto`:

<details><summary>full error log</summary>

```json
[
  {
    "id": "scrollable-region-focusable",
    "impact": "serious",
    "tags": [
      "cat.keyboard",
      "wcag2a",
      "wcag211",
      "wcag213",
      "TTv5",
      "TT4.a",
      "EN-301-549",
      "EN-9.2.1.1",
      "EN-9.2.1.3"
    ],
    "description": "Ensure elements that have scrollable content are accessible by keyboard",
    "help": "Scrollable region must have keyboard access",
    "helpUrl": "https://dequeuniversity.com/rules/axe/4.10/scrollable-region-focusable?application=axeAPI",
    "nodes": [
      {
        "any": [
          {
            "id": "focusable-content",
            "data": null,
            "relatedNodes": [],
            "impact": "serious",
            "message": "Element should have focusable content"
          },
          {
            "id": "focusable-element",
            "data": null,
            "relatedNodes": [],
            "impact": "serious",
            "message": "Element should be focusable"
          }
        ],
        "all": [],
        "none": [],
        "impact": "serious",
        "html": "<pre>\nconst popup = document.querySelector(\"#popup\"); \npopup.style.width = \"100%\";\npopup.innerText = \"Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.\";\n</pre>",
        "target": ["pre"],
        "failureSummary": "Fix any of the following:\n  Element should have focusable content\n  Element should be focusable"
      }
    ]
  }
]
```

</details>

At first I thought this was a weird error, because a screen reader could still read outloud the static inner text. But then I realized, no, someone might be not using a screen reader (relying on their vision), but using keyboard controls (e.g. motor disability), and won't be able to see the content.

So I tried putting the `tabindex` on the `<pre>` as suggested, but then eslint complains that I'm making a non-interactive HTML element interactive. (To work around this, I'm just disabling the eslint rule on `<pre>`s.)

So my question is... which rule is right? The eslint rule is also based on a WCAG standard, so it seems like there's some conflict here.

Feel free to close if this would be a better discussion on a WCAG repo somewhere, or on Stack Overflow.

References:
https://github.com/jsx-eslint/eslint-plugin-jsx-a11y/blob/cca288b73a39fa0932a57c02a7a88de68fc971fc/docs/rules/no-noninteractive-tabindex.md
https://accessibilityinsights.io/info-examples/web/scrollable-region-focusable/
https://github.com/rehype-pretty/rehype-pretty-code/issues/108

**Comment by WilcoFiers** [comment]
This is [`no-noninteractive-tabindex`](https://github.com/jsx-eslint/eslint-plugin-jsx-a11y/blob/main/docs/rules/no-noninteractive-tabindex.md), correct? I think this is a jsx-a11y false positive. WCAG requires all functionality to be keyboard operable, that includes scrolling. If there's nothing focusable inside a scroll container then the only way to do that is to make the scroll container itself focusable. There is no ARIA role for that. You could give the element an incorrect role to get jsx-a11y to pass, but assigning incorrect roles is a WCAG violation.

I think the best thing you can do is to turn that jsx-a11y rule off and report the false positive to jsx-a11y.
